chai-immutable.js

```diff
@@ -71,7 +71,7 @@ module.exports = function (chai, utils) {
     return function (collection) {
       var obj = this._obj;
 
-      if (obj && obj instanceof Collection) {
+      if (obj && obj instanceof Collection || collection instanceof Collection) {
```

Comment:
@matthewwithanm, I'm not sure I see why we need this.

Currently:

```js
expect([]).to.not.equal(List.of(1, 2, 3)); // Passes
expect([]).to.equal(List.of(1, 2, 3)); // Fails with "AssertionError: expected [] to equal List [ 1, 2, 3 ]"
```

which seems like what we should expect: if `obj` is not a `Collection`, the equality assertion is deferred to the original assertion.

Is there something I am I missing here?